src/pages/iou/steps/MoneyRequestAmountForm.js

```diff
@@ -0,0 +1,266 @@
+        setSelection({
+            start: amountAsStringForState.length,
+            end: amountAsStringForState.length,
+        });
+    };
+
+    useEffect(() => {
+        saveAmountToState(currency, amount);
+        // we want to update the state only when the amount is changed
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [amount]);
+
+    /**
+     * Sets the state according to amount that is passed
+     * @param {String} newAmount - Changed amount from user input
+     */
+    const setNewAmount = (newAmount) => {
+        // Remove spaces from the newAmount value because Safari on iOS adds spaces when pasting a copied value
+        // More info: https://github.com/Expensify/App/issues/16974
+        const newAmountWithoutSpaces = MoneyRequestUtils.stripSpacesFromAmount(newAmount);
+        // Use a shallow copy of selection to trigger setSelection
+        // More info: https://github.com/Expensify/App/issues/16385
+        if (!MoneyRequestUtils.validateAmount(newAmountWithoutSpaces)) {
+            setCurrentAmount((prevAmount) => prevAmount);
+            setSelection((prevSelection) => ({...prevSelection}));
+            return;
+        }
+        setCurrentAmount((prevAmount) => {
+            setSelection((prevSelection) => getNewSelection(prevSelection, prevAmount.length, newAmountWithoutSpaces.length));
+            return MoneyRequestUtils.stripCommaFromAmount(newAmountWithoutSpaces);
+        });
+    };
+
+    /**
+     * Update amount with number or Backspace pressed for BigNumberPad.
+     * Validate new amount with decimal number regex up to 6 digits and 2 decimal digit to enable Next button
+     *
+     * @param {String} key
+     */
+    const updateAmountNumberPad = useCallback(
+        (key) => {
+            if (shouldUpdateSelection && !textInput.current.isFocused()) {
+                textInput.current.focus();
+            }
+            // Backspace button is pressed
+            if (key === '<' || key === 'Backspace') {
+                if (currentAmount.length > 0) {
+                    const selectionStart = selection.start === selection.end ? selection.start - 1 : selection.start;
+                    const newAmount = `${currentAmount.substring(0, selectionStart)}${currentAmount.substring(selection.end)}`;
+                    setNewAmount(newAmount);
+                }
+                return;
+            }
+            const newAmount = MoneyRequestUtils.addLeadingZero(`${currentAmount.substring(0, selection.start)}${key}${currentAmount.substring(selection.end)}`);
+            setNewAmount(newAmount);
+        },
+        [currentAmount, selection, shouldUpdateSelection],
+    );
+
+    /**
+     * Update long press value, to remove items pressing on <
+     *
+     * @param {Boolean} value - Changed text from user input
+     */
+    const updateLongPressHandlerState = useCallback((value) => {
+        setShouldUpdateSelection(!value);
```

Comment:
**NAB/future consideration** but it would be great if we could somehow uncouple this selection logic. It feels like the `TextInputWithCurrencySymbol` should maybe be handling this stuff, but not sure if it's used anywhere else atm.